simba/schedule.py

```diff
@@ -476,6 +477,29 @@ def generate_scenario(self, args):
             start_simulation = datetime.datetime.fromtimestamp(0)
             stop_simulation = datetime.datetime.fromtimestamp(0)
 
+        daily = datetime.timedelta(days=1)
+        # read in time windows file
+        time_windows = None
+        if vars(args).get('time_windows'):
+            time_windows_path = Path(args.time_windows)
+            if time_windows_path.exists():
+                with time_windows_path.open('r', encoding='utf-8') as f:
+                    time_windows = util.uncomment_json_file(f)
+                # convert time window strings to date/times
+                for grid_operator, grid_operator_seasons in time_windows.items():
+                    for season, info in grid_operator_seasons.items():
+                        info["start"] = datetime.date.fromisoformat(info["start"])
+                        info["end"] = datetime.date.fromisoformat(info["end"])
+                        for level, windows in info.get("windows", {}).items():
+                            info["windows"][level] = [
+                                (datetime.time.fromisoformat(t[0]),
+                                 datetime.time.fromisoformat(t[1]))
+                                for t in windows]
+                        time_windows[grid_operator][season] = info
+                logging.debug("time windows read: " + str(time_windows))
+            else:
+                warnings.warn(f"Time windows file {args.time_windows} does not exist")
+
```

Comment:
can this be put in a function, e.g. parse_time_windows_from_file...?
This code is used elsewhere when time_windows are handled right?

Reply:
could be extracted to its own function. Problem: this is later done in SpiceEV, so it should be a SpiceEV function (which means waiting until this change has reached dev-branch). In addition, SpiceEV might also change the year if it does not correspond to the scenario time.